crates/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -400,31 +399,37 @@ impl PathEnvironment<Original> {
     /// Ensure that the path exists and contains files that "look" like an environment
     pub fn open(
         path: impl AsRef<Path>,
-        ident: EnvironmentRef,
         temp_dir: impl AsRef<Path>,
     ) -> Result<Self, EnvironmentError2> {
         let path = path.as_ref().to_path_buf();
         let dot_flox_path = path.join(".flox");
-        let env_path = dot_flox_path.join(ident.name().as_ref());
-
-        if !env_path.exists() {
-            Err(EnvironmentError2::EnvNotFound)?
-        }
-
-        if !env_path.is_dir() {
-            Err(EnvironmentError2::EnvNotADirectory)?
-        }
-
-        let env_path = env_path
+        let env_dir_entry = dot_flox_path
+            .read_dir()
+            .map_err(EnvironmentError2::ReadDotFlox)?
+            .find_map(|entry| {
+                let entry = entry.ok()?;
+                let path = entry.path();
+                if path.is_dir() {
+                    Some(entry)
+                } else {
+                    None
+                }
+            })
+            .ok_or(EnvironmentError2::EnvNotFound)?;
+
+        let env_path = env_dir_entry
+            .path()
             .canonicalize()
             .map_err(EnvironmentError2::EnvCanonicalize)?;
         if !env_path.join("flake.nix").exists() {
             Err(EnvironmentError2::DirectoryNotAnEnv)?
         }
 
+        let name = EnvironmentName::from_str(&env_dir_entry.file_name().to_string_lossy())?;
+
         Ok(PathEnvironment {
             path: env_path,
-            environment_ref: ident,
+            environment_ref: EnvironmentRef::new_from_parts(None, name),
```

Comment:
Can we stop using `EnvironmentRef` at this point?

Reply:
we can here, yes. the path environment only needs the environment name
its reasonable to use env ref for the other kinds of environments, for formatting and _referencing_ (in case of remote envs).

Reply:
We dont actually gain too much by only storing an `EnvironmentName`: 253fd8c9ece710aa302a1654d5cfb3308edf264c

I still think that change would make sense.